src/main/java/uk/ac/ebi/atlas/experiments/ExperimentJsonService.java

```diff
@@ -50,12 +50,13 @@ public JsonObject getExperimentJson(String experimentAccession, String accessKey
     }
 
     public ImmutableSet<JsonObject> getPublicExperimentsJson() {
-        // Sort by experiment type according to the above precedence list and then by display name
+        //Sort by experiment type according to the above precedence list and then by loadDate
```

Comment:
I would remove this comment and add its content to the method name to make it a well named method, something like this: `getPublicExperimentsByPrecedenceAndLoadDate`.

```suggestion
    public ImmutableSet<JsonObject> getPublicExperimentsByPrecedenceAndLoadDate() {
```

Reply:
I would like to ignore the tiny suggestion this time :) 
While I agree that clear naming is valuable, I’d argue that the method name should describe the result (what the caller gets), not the algorithm (how it’s built).
Including ordering in the method name makes the API brittle: if sorting rules change, we’d need to rename the method. A short comment is better for documenting that internal detail.

Reply:
I don't agree. What happens in your case when the sorting rule change? You do need to modify a long comment. If you forget it - and that is most of the cases - then the comment will be outdated and became useless.
I would say if the logic has totally changed inside your method, then you might want to change you method name. We do have editors nowadays that can handle method name's change easily.

Reply:
I think it is unnecessary  work to rename the function name just to refrain from using a comment. I don't have my editor setup to do this easily unfortunately. 
In case the sorting rule changes, we just update the comment. 
I understand your hatred towards comments, but I'm requesting for some leeway in this case.
It is a short comment after all and not something that is going to nuke the codebase :)